admin/class-acc_user_importer-admin.php

```diff
@@ -454,72 +354,73 @@ private function proccess_user_data ( $users ) {
+				}
+
+				// All done with updating the user
+				continue;
+			}
+
 			//--------USER NOT FOUND IN DATABASE-----
-			//But before creating a new record, make sure email is unique.
-			//We want emails to be unique in DB because it is a login identifier.
-			} elseif ( !(strlen($userEmail) > 1) ) {
+			// But before creating a new record, make sure email is valid.
+			if (!is_email($userEmail)) {
 				//User has no email field, skip it
-				$this->log_dual(" > error: no email given, cannot create new user account.");
+				$this->log_dual(" > error: no valid email given, cannot create new user account.");
 				$update_errors[] = $user;
-
-			} else {
-				$this->log_dual(" > user not found");
-
-				$user_id = email_exists($userEmail);
-				if ( is_numeric($user_id) ) {
-					//An existing user already has this email address, skip updating
-					$user2 = get_userdata($user_id);
-					$username2 = $user2->user_firstname . " " . $user2->user_lastname;
-					$this->log_dual(" > existing user #" . $user_id . " " . $username2 .
-					                " already has email " . $userEmail);
-					$this->log_dual(" > user update skipped");
-					$user_data = array_merge( array('ID' => $user_id), $user_data); //point to correct ID (future proofing)
-					$update_errors[] = $user;
-				} else {
-					//email is unique, proceed to create a new record
-					$update_this_user = true;
-					$this->log_dual(" > email not found on any other users");
-					$this->log_dual(" > will create new user account");
-					$new_users[] = $userContactId;
-					$new_users_email[] = $userEmail ;
-					$user_data["role"] = $default_role;
-				}
+				continue;
 			}
 
-			//only update user if needed
-			if ($update_this_user) {
-
-				//update core info (name, email)
-				$wp_user = wp_insert_user( $user_data );
-				if ( is_wp_error($wp_user) ) {
-					$this->log_dual(" > failed to update user");
-					$this->log_dual(" > WP:" . $wp_user->get_error_message());
-
-				} else {
-					//update user meta
-					update_user_meta( $wp_user, 'home_phone', $userHomePhone );
-					update_user_meta( $wp_user, 'cell_phone', $userCellPhone );
-					update_user_meta( $wp_user, 'membership', $userMembership );
-					update_user_meta( $wp_user, 'expiry', $userExpiry );
-					update_user_meta( $wp_user, 'city', $userCity );
-					$this->log_dual(" > Updated user information");
-					//Add user to the updated_users list only if it's not already in the created list
-					if (!in_array($userContactId, $new_users)) {
-						$updated_users[] = $userContactId;
-						$updated_users_email[] = $userEmail;
-					}
-				}
+			//--------CREATE NEW USER-----
+			$this->log_dual(" > email not found on any other users");
+			$this->log_dual(" > will create new user account");
+			$new_users[] = $userContactId;
+			$new_users_email[] = $userEmail ;
+			$accUserData["role"] = $default_role;
+			// Assign custom meta data
+			$accUserData["meta_input"] = $accUserMetaData;
```

Comment:
I noticed Wordpress has this core field :
'user_registered'      (string) Date the user registered. Format is 'Y-m-d H:i:s'.
just a thought, maybe we could eventually set this to today's date when we create a new user record.